bitrise-tools/codesigndoc
# Add UI Target support

**Description:**
For iOS Real Device Testing we need to generate an .xcterstrun file which was generated for generic/real device.
For this we need a development provisioning profile for the main target and for the UITest target too.

**Feature:**
Add a new scan option: xcodeuitests `./codesigndoc scan xcodeuitests` 

The scanner scans the provided project / workspace:
https://github.com/bitrise-tools/codesigndoc/pull/94/files#diff-977fdc092f258f73dc948cf69e0a2199R37

1) The scanner search for valid schemes for the provided iOS projectr / workspace.
  The scheme is valid if:
    - It has not skipped test target for test run:
      https://github.com/bitrise-tools/codesigndoc/pull/94/files#diff-afd3ea939b886556b2101c19dafba29bR174
<img width="893" alt="test" src="https://user-images.githubusercontent.com/37296013/51479327-3992e300-1d8e-11e9-8ca4-13ceeea90479.png">

    - There is a UITest target target in the not skipped ones :
      https://github.com/bitrise-tools/codesigndoc/pull/94/files#diff-afd3ea939b886556b2101c19dafba29bR182

  <img width="901" alt="enabled" src="https://user-images.githubusercontent.com/37296013/51479383-5af3cf00-1d8e-11e9-9484-27aa9d1b1409.png">


2) The user selects the scheme 

3) Run an xcodebuild build-for-testing command for the selected scheme:
    - https://github.com/bitrise-tools/codesigndoc/pull/94/files#diff-afd3ea939b886556b2101c19dafba29bR40

4) Export the codesigning files:
      https://github.com/bitrise-tools/codesigndoc/pull/94/files#diff-4d4b5fa386f84d41d927fb07965ea196R53
    - Check the builded artifacts for Test-Runner.app:
      https://github.com/bitrise-tools/codesigndoc/pull/94/files#diff-e5a906720c9a9db908dbc488cafdeefcR24

5) The user selects the codesigning files (certs & profiles) for every UITest target in the scheme:
https://github.com/bitrise-tools/codesigndoc/pull/94/files#diff-4d4b5fa386f84d41d927fb07965ea196R124


---

**How to test:**
You will need an iOS app with UITest enabled:
<img width="893" alt="test" src="https://user-images.githubusercontent.com/37296013/51479327-3992e300-1d8e-11e9-8ca4-13ceeea90479.png">

<img width="901" alt="enabled" src="https://user-images.githubusercontent.com/37296013/51479383-5af3cf00-1d8e-11e9-9484-27aa9d1b1409.png">

Run the `go run *.go scan xcodeuitests` from the root dir.

---
**Dependencies:**
Merge the dependendent PRs first:
https://github.com/bitrise-tools/xcode-project/pull/26 ✅
https://github.com/bitrise-tools/go-xcode/pull/72 ✅
